gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/task/COPCMessageHandler.java

```diff
@@ -100,8 +102,14 @@ public void checkAndMergeFileParts(InboundMessage inboundMessage, String convers
         Document ebXmlDocument = xPathService.parseDocumentFromXml(inboundMessage.getEbXML());
         var inboundMessageId = xPathService.getNodeValue(ebXmlDocument, MESSAGE_ID_PATH);
         var currentAttachmentLog = patientAttachmentLogService.findAttachmentLog(inboundMessageId, conversationId);
+        var canCheckAttachmentLength = false;
+
+        if (currentAttachmentLog != null) {
+            canCheckAttachmentLength = true;
+        }
 
-        if (currentAttachmentLog == null) {
+        // if a message has arrived early, it will not have a parent ID so we can cancel early.
+        if (currentAttachmentLog == null || currentAttachmentLog.getParentMid() == null) {
```

Comment:
There's a conflict here with the logic on line 123. Changes were brought in with NIAD-2156.